### PR TITLE
Decorative Flags (also handheld flags)

### PR DESCRIPTION
**Pull request by SaveliyM360**
## About the PR
I added flags to the game. It's a nice addition to the decor, and also allows servers to add their own faction and country lore flags, thanks to the existing blank flag base (hello from Baystation), such as Transsolar Federation, Space Wizard Federation, Moghes State, Arachnid Empire... I think you get the idea.

Of the variations of flags, there is the flag of NT, Syndicate, Pirates, LGBT, various color variations, as well as handheld (handy) versions for each of the flags.

## Why / Balance
I wrote above, and it's also not cool that there are banners in the game, but there are no flags.

## Technical details
They can be breaked just like posters. But unfortunately, they are available so far only in sandbox mode for mapping. Maybe in the future I will add them to the station maps (I hope it won't break anything).
Also, handy flags can be taken from the sandbox, and so far cannot be obtained in the game, as the big versions.

**_How to create your flag based on a blank flag sprite:_** Take the blank flag sprite, create a new layer under it, and set the blending mode to "multiple" on the blank flag layer. Before that, I recommend to select and cut on a new layer the sprites of "nails", which are located on the upper edges of the flag, and look like a "plus". 
(It work on paint.net, but you can check it yourself in Photoshop)

## Media

Flags_new2

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

- add: Added a large number of wall flags.  Now you can feel more patriotic by hanging your faction's flag on the wall.
- add: Added handheld flags in addition to large flags. Wave them around so everyone can see what a patriot you are.

### CODE REVIEW

File: Resources/Prototypes/Entities/Structures/Wallmounts/Signs/flags.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  description: "Glory to NT! Wait, they really made a flag for a corporation?"
+  components:
+  - type: Sprite
+    state: NT_flag
+
+- type: entity
+  parent: BaseFlag
+  id: SyndieFlag
+  name: Syndicate flag
+  description: "Smells bloody. Fuck NT!"
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Structures/Wallmounts/Signs/flags.yml:34]: that's a bad word

**Comment by deltanedas** [comment]
real

File: Resources/Prototypes/Entities/Structures/Wallmounts/Signs/flags.yml (hunk 1 above)
Comment (CatMagic1) [reply Resources/Prototypes/Entities/Structures/Wallmounts/Signs/flags.yml]: merge blocked

File: Resources/Textures/Structures/Wallmounts/flags.rsi/meta.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Taken from tgstation at https://github.com/tgstation/tgstation/commit/d0d81185f09ca30d3b0856d476544240dba0de53, drawed by Saveliy360",
```
Comment (Emisse) [inline Resources/Textures/Structures/Wallmounts/flags.rsi/meta.json:4]:
```suggestion
  "copyright": "Taken from tgstation at https://github.com/tgstation/tgstation/commit/d0d81185f09ca30d3b0856d476544240dba0de53, drawn by Saveliy360",
```

File: Resources/Textures/Structures/Wallmounts/flags.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Taken from tgstation at https://github.com/tgstation/tgstation/commit/d0d81185f09ca30d3b0856d476544240dba0de53, drawn by Saveliy360",
```
Comment (ficcialfaint) [inline Resources/Textures/Structures/Wallmounts/flags.rsi/meta.json:4]: doesn't seem too clear: what exactly was drawn by you and what was taken from /tg/

File: Resources/Prototypes/Entities/Structures/Wallmounts/Signs/flags.yml
Comment (ficcialfaint) [inline Resources/Prototypes/Entities/Structures/Wallmounts/Signs/flags.yml:1]: Can it spawn a destructed version of the flag as posters do?

**Comment by JoeHammad1844** [comment]
you forgot to mention you also added the french flag

File: Resources/Prototypes/Entities/Objects/Misc/handy_flags.yml [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    sprite: Objects/Misc/Handy_Flags/black_handy_flag.rsi
+    state: icon
+  - type: Item
+    sprite: Objects/Misc/Handy_Flags/black_handy_flag.rsi
+    
+- type: entity
+  parent: BaseItem
+  id: BlueHandyFlag
+  name: blue handy flag
+  description: "Just black handy flag"
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Misc/handy_flags.yml:77]: description is incorrect

File: Resources/Prototypes/Entities/Objects/Misc/handy_flags.yml [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    sprite: Objects/Misc/Handy_Flags/lgbtq_handy_flag.rsi
+    state: icon
+  - type: Item
+    sprite: Objects/Misc/Handy_Flags/lgbtq_handy_flag.rsi
+    
+- type: entity
+  parent: BaseItem
+  id: PirateHandyFlag
+  name: Pirate handy flag
+  description: "Holding it in your hands, show these carp that you're not kidding."
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Misc/handy_flags.yml:53]:
```suggestion
  id: PirateHandyFlag
  name: pirate handy flag
  description: Hold it in your hands to show these carp that you're not kidding.
```
descs dont need quotation marks around them + dont capitalize pirate imo
descriptions for most of these read pretty weirdly imo but ill just point them out as i go

File: Resources/Textures/Structures/Wallmounts/flags.rsi/meta.json [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+	{
+      "name": "lgbtq_flag"
+    },
+	{
+      "name": "pirate_flag"
+    },
+	{
+      "name": "black_flag"
+    },
+	{
```
Comment (mirrorcult) [inline Resources/Textures/Structures/Wallmounts/flags.rsi/meta.json:28]: lots of mixed tabs & spaces in this file, please only use spaces

File: Resources/Prototypes/Entities/Objects/Misc/handy_flags.yml [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    sprite: Objects/Misc/Handy_Flags/syndie_handy_flag.rsi
+    state: icon
+  - type: Item
+    sprite: Objects/Misc/Handy_Flags/syndie_handy_flag.rsi
+    
+- type: entity
+  parent: BaseItem
+  id: LGBTQHandyFlag
+  name: LGBTQ handy flag
+  description: "Be gay do crime handy flag"
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Misc/handy_flags.yml:41]:
```suggestion
  name: LGBTQ handy flag
  description: "The be gay do crime handy flag."
```

all flag descs here need periods at the end of them, the solid color flags mostly

also think this desc reads better like this, same for the wallmount flag

File: Resources/Prototypes/Entities/Objects/Misc/handy_flags.yml [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+    sprite: Objects/Misc/Handy_Flags/pirate_handy_flag.rsi
+    state: icon
+  - type: Item
+    sprite: Objects/Misc/Handy_Flags/pirate_handy_flag.rsi
+    
+- type: entity
+  parent: BaseItem
+  id: BlackHandyFlag
+  name: black handy flag
+  description: "Just black handy flag"
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Misc/handy_flags.yml:65]:
```suggestion
  id: BlackHandyFlag
  name: black handy flag
  description: Just a black handy flag.
```

same for all the other color flags, not going to review them separately

**Review by mirrorcult — CHANGES_REQUESTED** [review]
the solid color flags can very easily use a grayscale layer that has its color specified in the sprite component instead, rather than duplicating so many images like this. the other ones are fine to be unique. (maybe even just try setting color on the gray solid color flag and see how it looks?)

"handy flag" reads kind of weird to me? ive never heard them called that before in english, i would call them "handheld flags"

**Comment by SaveliyM360** [comment]
> Ok, I think I fixed all the typos in the files, and renamed all the handy things to handheld. About grayscale mechanics for the flags, I don't really understand it. When I creating colored flags, I focused on the color variety of bedspreads, because I thought that five flags would be too few.
If you mean to remove the colored flags, then I will.

**Comment by mirrorcult** [comment]
you can just have one grayscaled sprite for the flags, and then color them dynamically in YAML by changing the `color:` field in spritecomponent

**Comment by SaveliyM360** [comment]
> you can just have one grayscaled sprite for the flags, and then color them dynamically in YAML by changing the `color:` field in spritecomponent

Oh, i got it. Well, in fact, this role is performed by the blank flag.

**Comment by SaveliyM360** [comment]
Okay, now all coloured flags are removed.

**Comment by SaveliyM360** [comment]
I replaced the picture, due to the removal of colored flags.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
"Drawn by" not "drawed by" please, also faint's review.

**Pull request merged**